static/js/stream_settings_ui.js

```diff
@@ -299,11 +299,8 @@ export function remove_stream(stream_id) {
 export function update_settings_for_subscribed(slim_sub) {
     const sub = stream_settings_data.get_sub_for_settings(slim_sub);
     stream_ui_updates.update_add_subscriptions_elements(sub);
-    $(
-        `.subscription_settings[data-stream-id='${CSS.escape(
-            sub.stream_id,
-        )}'] #preview-stream-button`,
-    ).show();
+    $(`.subscription_settings[data-stream-id='${CSS.escape(sub.stream_id)}']`).show();
```

Comment:
I don't understand this change; the previous selector should match the preview-stream-button, if the stream in question is already displayed.

I suspect the actual bug may be that we are doing this too early, and it should be moved after the `is_sub_already_present` block.

Reply:
@timabbott , I tried moving it after is_sub_already_present and it doesn't work. It's funny how the above fix actually works. I assume ``` $(`#preview-stream-button`).show(); ``` , didn't get rendered first time.